lib/renuo_bin_check/bin_check.rb

```diff
@@ -0,0 +1,61 @@
+class BinCheck
+  def self.method_missing(name, *_params, &configs)
+    if block_given?
+      @current_name = name
+      @new_script_hash[@current_name] = {}
+      instance_eval(&configs)
+    else
+      super
+    end
+  end
+
+  def respond_to_missing?
+    #:nocov:
+    true
+    #:nocov:
+  end
+
+  def self.run(&check)
+    @new_script_hash = {}
+    instance_eval(&check)
+    initialize_checks
+  end
+
+  #:reek:NestedIterators
+  def self.initialize_checks
+    initializer = RenuoBinCheck::Initializer.new
+    @new_script_hash.each do |name, configs|
+      initializer.check do |config|
+        config.name name
+        configs.each do |key, value|
```

Comment:
woud it make sense to pack the nested iterator to a seperate class so we could get rid of the `#:reek:NestedIterators` ?